### PR TITLE
New progress indicator and cancel button

### PR DESCRIPTION
**Pull request by mickel1138**
Replaced loading panel with custom popup. With more info and a button.

### CODE REVIEW

File: assets/js/vue/FileManager.vue [hunk 1]
```diff
@@ -25,6 +15,46 @@
...
+                />
+                <center>
+                <DxButton
+                    text="Cancel Upload!"
+                    type="danger"
+                    styling-mode="contained"
+                    @click="stopProcess"
+                 />
+                 <p>
+                    If the filesize is large? Click Cancel!
```
Comment (mickel1138) [inline assets/js/vue/FileManager.vue:53]: This message is more of a placeholder. It should appear maybe when large file size is detected?

File: assets/js/vue/FileManager.vue [hunk 2]
```diff
@@ -25,6 +15,46 @@
...
+            :close-on-outside-click="false"
+            :show-title="false"
+            position="center"
+            :showCloseButton="false"
+            :width="300"
+            :height="250"
+            :drag-enabled="false"
+        >
+            <template>
+                <center>
```
Comment (praneethpr) [inline assets/js/vue/FileManager.vue:30]: <center> tag is deprecated

File: assets/js/vue/FileManager.vue (hunk 1 above)
Comment (praneethpr) [reply assets/js/vue/FileManager.vue:53]: I think the generic message is good, we can ask the PO about what they think

File: assets/js/vue/FileManager.vue [hunk 3]
```diff
@@ -83,10 +114,21 @@ let fileManagerResolve = [];
...
 let doneFiles = 0;
 
+let totalFileSize = 0;
+let doneFileSize = 0;
+
 let myFileManager;
-let myLoadPanel;
 let myDropzone;
 
+let mags = ' KMGTPEZY';
```
Comment (praneethpr) [inline assets/js/vue/FileManager.vue:123]: Could be moved to a util function

File: assets/js/vue/FileManager.vue [hunk 4]
```diff
@@ -83,10 +114,21 @@ let fileManagerResolve = [];
 let totalFiles = 0;
```
Comment (praneethpr) [inline assets/js/vue/FileManager.vue:114]: Can remove itemsChanged variable, can move totalFiles and doneFiles inside initDropzone()

File: assets/js/vue/FileManager.vue [hunk 5]
```diff
@@ -83,10 +114,21 @@ let fileManagerResolve = [];
...
 let doneFiles = 0;
 
+let totalFileSize = 0;
```
Comment (praneethpr) [inline assets/js/vue/FileManager.vue:117]: We need to get rid of these variables in global scope

File: assets/js/vue/FileManager.vue [hunk 6]
```diff
@@ -407,13 +458,17 @@ const initDropzone = () => {
...
+        myFileManager.$parent.doneFileSize += file.size;
+        myFileManager.$parent.uploadMessage = "Uploading " + doneFiles + " of " + totalFiles;
+        myFileManager.$parent.bytesMessage = "Uploading " + humanSize(myFileManager.$parent.doneFileSize) + " of " + humanSize(myFileManager.$parent.totalFileSize);
     });
 
     myDropzone.on("queuecomplete", function () {
         myFileManager.instance.repaint();
         totalFiles = 0;
         doneFiles = 0;
+        totalFileSize = 0;
```
Comment (praneethpr) [inline assets/js/vue/FileManager.vue:470]: We need this to point to myFileManager.$parent.doneFileSize and myFileManager.$parent.totalFileSize

File: assets/js/vue/FileManager.vue [hunk 7]
```diff
@@ -360,6 +409,7 @@ const initDropzone = () => {
         maxFilesize: null,
         clickable: "#upload-file-button",
         timeout: 0,
+        autoProcessQueue: true,
```
Comment (praneethpr) [inline assets/js/vue/FileManager.vue:412]: By default it is true, we can remove this

File: assets/js/vue/FileManager.vue [hunk 8]
```diff
@@ -83,10 +114,21 @@ let fileManagerResolve = [];
...
+let totalFileSize = 0;
+let doneFileSize = 0;
+
 let myFileManager;
-let myLoadPanel;
 let myDropzone;
 
+let mags = ' KMGTPEZY';
+
+function humanSize(bytes, precision) {
```
Comment (praneethpr) [inline assets/js/vue/FileManager.vue:125]: We could also use our existing function in TwigExtension::formatBytes()

File: assets/js/vue/FileManager.vue (hunk 2 above)
Comment (mickel1138) [reply assets/js/vue/FileManager.vue:30]: I will become a div with a class and styling.

File: assets/js/vue/FileManager.vue (hunk 8 above)
Comment (mickel1138) [reply assets/js/vue/FileManager.vue:125]: hmmm, would be consistent...

**Review by praneethpr — APPROVED** [review]

**Pull request merged**